packages/filesystem/src/node/node-filesystem.ts

```diff
@@ -387,36 +335,41 @@ export class FileSystemNode implements FileSystem {
         }
     }
 
-    protected doGetEncoding(option?: { encoding?: string }): string {
-        return option && typeof (option.encoding) !== "undefined"
+    protected async doGetEncoding(option?: { encoding?: string }): Promise<string> {
```

Comment:
Why is this async and Promise Return ? 
Smae for many option dependent methods below...

Reply:
Subclasses my get the encoding (and others) differently. At least that was my idea.

Reply:
I would think that the possibly async method would be the one to request options and then what handles the options could be sync since it's just a helper to data. WDYT ?

Reply:
I am fine with the current changes.